lib/oban/query.ex

```diff
@@ -70,11 +70,25 @@ defmodule Oban.Query do
   end
 
   @spec stage_scheduled_jobs(Config.t(), binary()) :: {integer(), nil}
-  def stage_scheduled_jobs(%Config{prefix: prefix, repo: repo, verbose: verbose}, queue) do
+  def stage_scheduled_jobs(%Config{} = config, queue) do
+    stage_scheduled_jobs(config, queue, fn queryable ->
+      where(queryable, [j], j.scheduled_at <= ^utc_now())
+    end)
+  end
+
+  @spec stage_all_scheduled_jobs(Config.t(), binary()) :: {integer(), nil}
+  def stage_all_scheduled_jobs(%Config{} = config, queue) do
```

Comment:
It doesn't seem right to have this extra function between `stage_scheduled_jobs/2` and `stage_scheduled_jobs/3`. The standard behavior is to check the `scheduled_at` value (which every queue does once a second), it shouldn't be a special case. The same compound index will always be used to build the query regardless of whether `scheduled_at` is checked or not.

What if the "far future" value was dynamically calculated at compile time rather than a fixed number? It makes no sense to schedule a job 10 years in the future (or one year for that matter).

Reply:
> The same compound index will always be used to build the query regardless of whether scheduled_at is checked or not.

🤔 Yeah, but it would still be more expensive to compare the timestamps while traversing the index rather than just take all after the `state` check?

Nonetheless, I changed it back to a far future timestamp (compile time, as before) and removed `stage_all_scheduled_jobs`.

👍 Thanks again for your reviews! Very insightful, I learned a bunch!